app/models/error_detail.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class ErrorDetail < ApplicationRecord
+  include Discard::Model
+  self.discard_column = :deleted_at
+
+  belongs_to :integration, polymorphic: true, optional: true
```

Comment:
relationship to integration should not be polymorphic. For integrations we use STI, and relationship to the integration is actually relationship to the `Integrations::BaseIntegration` model

Reply:
the thing is that we also want to store payment_providers here, and we chose :integration as polymorphic relation to highlight that if something in our codebase is triggering integration error, most likely it should follow pattern of integrations

Reply:
I think that the answer lies in the decision how we are going to use it: [https://github.com/getlago/lago-api/pull/2316/files#r1687997757](url)
1. If we will have something generic in `error_code` like `tax_error` or `payment_provider_error`, then we can completely remove this association I guess. It is easy to get the integration source: `invoice.customer.anrok_customer.integration` and we already have organization and owner relationship. In that case for `details` we will have `{tax_error: 'xyz'}`. If multiple errors are attached to the one invoice we can easily filter them by `error_code`.
2. If specific error code will be written in `error_code` field (like `taxDateTooFarInFuture`), in that case maybe it is useful to have integration relationship and I guess it can stay polymorphic, even I don't see in the near future we will use it for sth other than integration. In this case, at least for Anrok, we won't be using details field but it is fine I guess

Reply:
yep, just clarified: an organization can have together multiple connections with both xero and netsuite, but on the customer level we'll have only one

Reply:
exactly, on customer level you cannot be connected to multiple tax providers, because you need only one source of truth. For payment providers it should work the same. One customer cannot have multiple payment providers attached (that would mean that he/she would be billed multiple times)